apps/camera_pipe/camera_pipe.cpp

```diff
@@ -279,50 +248,32 @@ Func process(Func raw, Type result_type,
     Expr out_width = processed.output_buffer().width();
     Expr out_height = processed.output_buffer().height();
 
-    processed.bound(c, 0, 3); // bound color loop 0-3, properly
-    if (target.arch == Target::ARM) {
-        // Compute in chunks over tiles, vectorized by 8
-        const int tile_size = 32;
-        denoised.compute_at(processed, tx)
-            .vectorize(x, 8);
-        deinterleaved.compute_at(processed, tx)
-            .vectorize(x, 8)
-            .reorder(c, x, y)
-            .unroll(c);
-        corrected.compute_at(processed, tx)
-            .vectorize(x, 4)
-            .reorder(c, x, y)
-            .unroll(c);
-        processed.compute_root()
-            .tile(x, y, tx, ty, xi, yi, tile_size, tile_size)
-            .reorder(xi, yi, c, tx, ty)
-            .parallel(ty);
-
-        // We can generate slightly better code if we know the output is a whole number of tiles.
-        processed
-            .bound(x, 0, (out_width/tile_size)*tile_size)
-            .bound(y, 0, (out_height/tile_size)*tile_size);
-    } else if (target.arch == Target::X86) {
-        // Same as above, but don't vectorize (sse is bad at interleaved 16-bit ops)
-        const int tile_size = 128;
-        denoised.compute_at(processed, tx);
-        deinterleaved.compute_at(processed, tx);
-        corrected.compute_at(processed, tx);
-        processed.compute_root()
-            .tile(x, y, tx, ty, xi, yi, tile_size, tile_size)
-            .reorder(xi, yi, c, tx, ty)
-            .parallel(ty);
-
-        // We can generate slightly better code if we know the output is a whole number of tiles.
-        processed
-            .bound(x, 0, (out_width/tile_size)*tile_size)
-            .bound(y, 0, (out_height/tile_size)*tile_size);
-    } else {
-        denoised.compute_root();
-        deinterleaved.compute_root();
-        corrected.compute_root();
-        processed.compute_root();
-    }
+    const int strip_size = 32;
```

Comment:
128 is much slower, funnily enough

On Wed, Jun 1, 2016 at 1:16 PM, Dillon Sharlet <notifications@github.com>
wrote:

> In apps/camera_pipe/camera_pipe.cpp
> <https://github.com/halide/Halide/pull/1262#discussion_r65434086>:
>
> > -        processed.compute_root()
> > -            .tile(x, y, tx, ty, xi, yi, tile_size, tile_size)
> > -            .reorder(xi, yi, c, tx, ty)
> > -            .parallel(ty);
> > -
> > -        // We can generate slightly better code if we know the output is a whole number of tiles.
> > -        processed
> > -            .bound(x, 0, (out_width/tile_size)*tile_size)
> > -            .bound(y, 0, (out_height/tile_size)*tile_size);
> > -    } else {
> > -        denoised.compute_root();
> > -        deinterleaved.compute_root();
> > -        corrected.compute_root();
> > -        processed.compute_root();
> > -    }
> > +    const int strip_size = 32;
>
> I think we should leave this as 128 (the old tile size), it should reduce
> redundant compute without decreasing memory locality.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/halide/Halide/pull/1262/files/cdc1b77c2ce4e953b93307ac669d3bce76f020ea#r65434086>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAfdRkFR8mZOZvQTB9d36O83MqnauBw-ks5qHeiygaJpZM4Ir6sp>
> .
>